client/src/main/kotlin/tech/figure/validationoracle/client/domain/serialization/ValidationDefinitionDeletionRequestSerializer.kt

```diff
@@ -0,0 +1,31 @@
+package tech.figure.validationoracle.client.domain.serialization
+
+import com.fasterxml.jackson.core.JsonGenerator
+import com.fasterxml.jackson.databind.JsonSerializer
+import com.fasterxml.jackson.databind.SerializerProvider
+import tech.figure.validationoracle.client.domain.execute.ValidationDefinitionDeletionRequest
+
+/**
+ * The JSON structure of a contract entry point is impossible to represent with vanilla Jackson annotation setups. This
+ * serializer enables the object to be serialized correctly with its multiple-nested nodes without enabling special
+ * ObjectMapper features, allowing the object to be more universally applicable to external ObjectMapper singleton
+ * instances.
+ *
+ * This serializer outputs the values in the following format:
+ * ```json
+ * {
+ *      "delete_validation_definition": {
+ *          "validation_type": ...
+ *      }
+ * }
+ * ```
+ */
+class ValidationDefinitionDeletionRequestSerializer : JsonSerializer<ValidationDefinitionDeletionRequest>() {
+    override fun serialize(value: ValidationDefinitionDeletionRequest, gen: JsonGenerator, provider: SerializerProvider?) {
+        gen.writeStartObject() // Start root node
+        gen.writeObjectFieldStart("delete_validation_definition") // Start delete_validation_definition node
+        gen.writeStringField("validation_type", value.validationType)
+        gen.writeEndObject() // End delete_validation_definition node
+        gen.writeEndObject() // End root node
```

Comment:
Any thought to a quick DSL to make this less error prone?

Disclaimer: 100% untested code (that "compiles" in a scratch file) for an idea of what I was thinking:
```kotlin
class Generator{
    fun writeStartObject() {}
    fun writeObjectFieldStart(s: String) {}
    fun <V> writeStringField(s: String, value: V) {}
    fun writeEndObject() {}
}
val gen = Generator()
// ^ dummy object

gen.writeStartObject() // Start root node
gen.writeObjectFieldStart("delete_validation_definition") // Start delete_validation_definition node
gen.writeStringField("validation_type", true)
gen.writeEndObject() // End delete_validation_definition node
gen.writeEndObject() // End root node

// ^ OG way


fun Generator.root(fn: Generator.() -> Unit): Generator = this.also {
    writeStartObject() // Start root node
    fn()
    writeEndObject() // End root node
}
fun Generator.obj(s: String, fn: Generator.() -> Unit): Generator = this.also {
    gen.writeObjectFieldStart(s) // Start delete_validation_definition node
    fn()
    gen.writeEndObject() // End delete_validation_definition node
}

gen.root { 
    obj("delete_validation_definition") {
        writeStringField("key", true)
    }
}

// ^ groovy dsl
```

Reply:
I do hope the integration tests will eventually be validating these serializers, but this is a good idea regardless, so will try doing this.

Reply:
No one likes holding virtual stacks in their brain